items:
- uid: com.microsoft.rest._rest_exception
  id: _rest_exception
  parent: com.microsoft.rest
  children:
  - com.microsoft.rest._rest_exception_1a8b86502c682e0b0e47f60b1c33b67b8f
  - com.microsoft.rest._rest_exception_1a8a0b21d2d492219e29a07567ff52b9cf
  - com.microsoft.rest._rest_exception_1ae3378da3ede0800826ccfe5f68532460
  - com.microsoft.rest._rest_exception_1ac00611a5b9a02b99e257f2407dc679d5
  href: com.microsoft.rest._rest_exception.yml
  langs:
  - java
  name: RestException
  fullName: com.microsoft.rest.RestException
  type: Class
  source:
    remote: &o0
      path: runtimes/client-runtime/src/main/java/com/microsoft/rest/RestException.java
      branch: master
      repo: https://github.com/Azure/azure-sdk-for-java.git
    path: runtimes/client-runtime/src/main/java/com/microsoft/rest/RestException.java
    startLine: 12
  package: com.microsoft.rest
  summary: <p>Exception thrown for an invalid response with custom error information. </p>
  syntax:
    content: public class RestException
  inheritance:
  - Exception
- uid: com.microsoft.rest._rest_exception_1a8b86502c682e0b0e47f60b1c33b67b8f
  id: _rest_exception_1a8b86502c682e0b0e47f60b1c33b67b8f
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  langs:
  - java
  name: RestException()
  fullName: com.microsoft.rest.RestException.RestException()
  type: Method
  source:
    remote: *o0
    path: runtimes/client-runtime/src/main/java/com/microsoft/rest/RestException.java
    startLine: 16
  package: com.microsoft.rest
  summary: <p>Initializes a new instance of the AutoRestException class. </p>
  syntax:
    content: public RestException()
- uid: com.microsoft.rest._rest_exception_1a8a0b21d2d492219e29a07567ff52b9cf
  id: _rest_exception_1a8a0b21d2d492219e29a07567ff52b9cf
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  langs:
  - java
  name: RestException(String message)
  fullName: com.microsoft.rest.RestException.RestException(String message)
  type: Method
  source:
    remote: *o0
    path: runtimes/client-runtime/src/main/java/com/microsoft/rest/RestException.java
    startLine: 23
  package: com.microsoft.rest
  summary: >-
    <p>Initializes a new instance of the AutoRestException class.</p>

    <p></p>
  syntax:
    content: public RestException(String message)
    parameters:
    - id: message
      type: "26831127"
      description: <p>The exception message. </p>
- uid: com.microsoft.rest._rest_exception_1ae3378da3ede0800826ccfe5f68532460
  id: _rest_exception_1ae3378da3ede0800826ccfe5f68532460
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  langs:
  - java
  name: RestException(String message, Throwable cause)
  fullName: com.microsoft.rest.RestException.RestException(String message, Throwable cause)
  type: Method
  source:
    remote: *o0
    path: runtimes/client-runtime/src/main/java/com/microsoft/rest/RestException.java
    startLine: 42
  package: com.microsoft.rest
  summary: >-
    <p>Initializes a new instance of the AutoRestException class.</p>

    <p></p>
  syntax:
    content: public RestException(String message, Throwable cause)
    parameters:
    - id: message
      type: "26831127"
      description: <p>the exception message </p>
    - id: cause
      type: a04fafd7
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.rest._rest_exception_1ac00611a5b9a02b99e257f2407dc679d5
  id: _rest_exception_1ac00611a5b9a02b99e257f2407dc679d5
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  langs:
  - java
  name: RestException(Throwable cause)
  fullName: com.microsoft.rest.RestException.RestException(Throwable cause)
  type: Method
  source:
    remote: *o0
    path: runtimes/client-runtime/src/main/java/com/microsoft/rest/RestException.java
    startLine: 32
  package: com.microsoft.rest
  summary: >-
    <p>Initializes a new instance of the AutoRestException class.</p>

    <p></p>
  syntax:
    content: public RestException(Throwable cause)
    parameters:
    - id: cause
      type: a04fafd7
      description: <p>exception that caused this exception to occur </p>
references:
- uid: "26831127"
  spec.java:
  - name: String
    fullName: String
- uid: a04fafd7
  spec.java:
  - name: Throwable
    fullName: Throwable
- uid: com.microsoft.azure.batch.protocol.models._batch_error_exception
  parent: com.microsoft.azure.batch.protocol.models
  href: com.microsoft.azure.batch.protocol.models._batch_error_exception.yml
  name: BatchErrorException
  fullName: com.microsoft.azure.batch.protocol.models.BatchErrorException
  type: Class
  summary: <p>Exception thrown for an invalid response with <xref href="com.microsoft.azure.batch.protocol.models._batch_error" data-throw-if-not-resolved="false"></xref> information. </p>
  syntax:
    content: public class BatchErrorException
- uid: com.microsoft.azure._cloud_exception
  parent: com.microsoft.azure
  href: com.microsoft.azure._cloud_exception.yml
  name: CloudException
  fullName: com.microsoft.azure.CloudException
  type: Class
  summary: <p>Exception thrown for an invalid response with custom error information. </p>
  syntax:
    content: public class CloudException
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  parent: com.microsoft.azure.management.datalake.store.models
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  name: AdlsErrorException
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsErrorException
  type: Class
  summary: <p>Exception thrown for an invalid response with <xref href="com.microsoft.azure.management.datalake.store.models._adls_error" data-throw-if-not-resolved="false"></xref> information. </p>
  syntax:
    content: public class AdlsErrorException
- uid: com.microsoft.rest._service_exception
  parent: com.microsoft.rest
  href: com.microsoft.rest._service_exception.yml
  name: ServiceException
  fullName: com.microsoft.rest.ServiceException
  type: Class
  summary: <p>Exception thrown for an invalid response with custom error information. </p>
  syntax:
    content: public class ServiceException
